Find a functionally equivalent code:com.android.keyguard.KeyguardTransportControlView.populateMetadata:COMMENT
Method Modifier: private     
Comment:/**
 * Populates the given metadata into the view
 */

@@ -10,19 +10,17 @@
         Log.e(TAG, "Couldn't get remote control client package icon", e);
     }
     setBadgeIcon(badgeIcon);
-    if (!TextUtils.isEmpty(mMetadata.trackTitle)) {
-        mTrackTitle.setText(mMetadata.trackTitle);
-    }
-    StringBuilder sb = new StringBuilder();
+    mTrackTitle.setText(!TextUtils.isEmpty(mMetadata.trackTitle) ? mMetadata.trackTitle : null);
+    final StringBuilder sb = new StringBuilder();
     if (!TextUtils.isEmpty(mMetadata.artist)) {
         if (sb.length() != 0) {
             sb.append(" - ");
@@ -26,32 +24,31 @@
         }
         sb.append(mMetadata.albumTitle);
     }
-    mTrackArtistAlbum.setText(sb.toString());
+    final String trackArtistAlbum = sb.toString();
+    mTrackArtistAlbum.setText(!TextUtils.isEmpty(trackArtistAlbum) ? trackArtistAlbum : null);
     if (mMetadata.duration >= 0) {
         setSeekBarsEnabled(true);
         setSeekBarDuration(mMetadata.duration);

